src/tools/pub/package-layout.md

```diff
@@ -128,11 +128,51 @@ enchilada/
   CHANGELOG.md
 {% endprettify %}
 
-To show users the latest changes to your package, you can include a changelog
-file where you can write a short note about the changes in your latest
-release. When you upload your package to the
-[pub.dev site,]({{site.pub}}) your package's `CHANGELOG.md` file (if any)
-appears in the changelog tab, rendered as [Markdown.][Markdown]
+Include a `CHANGELOG.md` file that has a section for
+each release of your package,
+with notes to help users of your package upgrade.
+Users of your package often review the changelog
+to discover bug fixes and new features,
+or to determine how much effort it will take to upgrade
+to the latest version of your package.
+
+To support tools that parse `CHANGELOG.md`,
+use the following format:
+
+* Each version has its own section with a heading.
+* The version headings are either all level 1 or all level 2.
+* The version heading text contains a package version number,
+  optionally prefixed with "v".
+
+When you upload your package to the [pub.dev site,]({{site.pub}})
+your package's `CHANGELOG.md` file (if any)
+appears in the **Changelog** tab, rendered as [Markdown.][Markdown]
+
+Here's an example of a `CHANGELOG.md` file.
+As the example shows, you can add subsections.
+
+{% prettify markdown tag=pre+code %}
+# 1.0.1
+
+* Fixed missing exclamation mark in `sayHi()` method.
+
+# 1.0.0
+
+* **Breaking change:** Removed deprecated `sayHello()` method.
```

Comment:
Perhaps we should move the example of a breaking change from v1.0.0 to v2.0.0.
I think there is a strong tendency for v1.0.0 to be a 'graduation' of a 0.x.y version without further change.

Reply:
Although 0.1.1 did deprecate sayHello...